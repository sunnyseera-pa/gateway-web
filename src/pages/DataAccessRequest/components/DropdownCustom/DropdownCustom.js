import React from 'react'; 
import { Dropdown } from 'react-bootstrap';
import axios from 'axios';
import _ from 'lodash';
import './DropdownCustom.scss';
 
let baseURL = require('../../../commonComponents/BaseURL').getURL();

export const CustomToggle = React.forwardRef(({ children, onClick }, ref) => (
    <a
        className='noTextDecoration'
        href='javascript:void(0)'
        ref={ref}
        onClick={e => {
            e.preventDefault();
                onClick(e);
        }}>
        {children}
    </a>
)); 

class DropdownCustom extends React.Component {
       // initialize our state
	state = { 
        darApplicationId: '', 
		contributorsInfo: [],
	};

    constructor(props) {
        super(props); 
        this.state = {
            value: this.props.value,
            readOnly: props.readOnly || false,
        };
        this.handleFocus = this.handleFocus.bind(this);
		this.handleBlur = this.handleBlur.bind(this);
        this.state.darApplicationId = props.applicationId;
    }

    async componentDidMount() {
		await this.getContributorsInfo();
	}

    async getContributorsInfo() {
        await axios.get(baseURL + `/api/v1/data-access-request/prepopulate-contributors/${this.state.darApplicationId}`).then(res => {
            this.setState({
                contributorsInfo: res.data.data
            });
        });
    }

    setContributorValue(contributor) {
        let contributorValue = `${contributor.firstname} ${contributor.lastname}`

        this.setState(
            {
                value: contributorValue,
            },
            this.props.onChange.bind(null, contributor)
        );

    }

    handleChange(e) {
        this.setState(
            {
                value: e.target.value,
            },
            this.props.onChange.bind(null, e.target.value)
        );
    }

    handleFocus(e) {
		this.props.onFocus();
	}
	handleBlur(e) {
		this.props.onBlur(this.props.value);
	}

    render() {
        const { contributorsInfo } = this.state;
        
        return (      
           <Dropdown >
					<Dropdown.Toggle as={CustomToggle}> 
                        <input 
                            name={this.props.name}
                            id={this.props.id}
					        disabled={this.state.readOnly}
                            aria-labelledby={this.props.labelId}
					        className={this.props.classes.input}
					        required={this.props.required ? 'required' : undefined}
                            type="text" 
							onChange={e => this.handleChange(e)} 
                            onBlur={this.handleBlur}
				            onFocus={this.handleFocus}
                            value={this.state.value}  
                            data-test-id='darContributorTextInput'
                        ></input> 
					</Dropdown.Toggle>
					<Dropdown.Menu className='darContributorsDropdown' >
                    <span className='gray800-14-bold'> Suggestions: </span>
						{!_.isUndefined(contributorsInfo) && contributorsInfo.map((contributor, index) => { 
							return (
                                <div 
                                    className='margin-top-8 cursorPointer' 
                                    data-test-id='darContributorDropdownItem' key={index}
                                    onClick={() => {
                                        this.setContributorValue(contributor)
                                    }}
                                >
                                    <span className='gray800-14'>{contributor.firstname} {contributor.lastname} {!_.has(contributor, 'user.email') && ' (contributor)'}</span>
                                    <br />
                                    <span className='gray-600-14' data-test-id={`darContributorDropdownEmail${index}`}>
                                        {_.has(contributor, 'user.email') ? contributor.user.email : 'Email address cannot be shared'}
                                    </span>
                                    <span className='gray-600-14 floatRight' data-test-id={`darContributorDropdownOrganisation${index}`}>
                                        {!_.has(contributor, 'user.email') && contributor.showOrganisation === false ? 'Organisation is hidden' : contributor.organisation}
                                    </span>
                                </div>
							); 
						})}
					</Dropdown.Menu>
				</Dropdown>
        );
    }
}

DropdownCustom.defaultProps = {
    value: '',
	onChange: () => {},
	onFocus: () => {},
	onBlur: () => {},
};

export default DropdownCustom;
