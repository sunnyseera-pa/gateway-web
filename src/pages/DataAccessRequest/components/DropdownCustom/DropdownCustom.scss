@import '../../../../css/variables';
@import '../../../../css/typography';

.darContributorsDropdown {
    width: 100%;
    padding: 8px 16px !important;
}

.noTextDecoration,
.noTextDecoration:hover  {
	text-decoration: none;
}