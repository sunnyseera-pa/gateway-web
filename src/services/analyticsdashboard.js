import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
