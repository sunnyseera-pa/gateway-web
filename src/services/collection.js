import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
