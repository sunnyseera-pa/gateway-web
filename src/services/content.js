import { useQuery } from 'react-query';
import { getRequest } from '../utils/requests';
import { UATCMSURL } from './content.constants';

const baseURL = require('../pages/commonComponents/BaseURL');
const cmsURL = baseURL.getCMSURL();
const env = baseURL.getURLEnv();
const local = 'local';

export const getNon5SafesModalContentRequest = (options) => {
    const url = env === local ? UATCMSURL : cmsURL;
	return getRequest(`${url}/Non5SafesModalContent`, options);
};

const useGetNon5SafesModalContentRequest = (requestOptions, queryOptions = { queryKey: 'getNon5SafesModalContentRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getNon5SafesModalContentRequest(requestOptions),
	});
};

export default {
    useGetNon5SafesModalContentRequest
}
