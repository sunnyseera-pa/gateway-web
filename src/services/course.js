import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest, putRequest } from '../utils/requests';

const getCourse = (_id, options) => {
	return getRequest(`${apiURL}/course/${_id}`, options);
};

const getEdit = (_id, options) => {
	return getRequest(`${apiURL}/course/edit/${_id}`, options);
};

const postCourse = (data, options) => {
	return postRequest(`${apiURL}/course`, data, options);
};

const putCourse = (_id, data, options) => {
	return putRequest(`${apiURL}/course/${_id}`, data, options);
};

const useGetCourse = (requestOptions, queryOptions = { queryKey: 'getCourse' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getCourse(_id, requestOptions),
	});
};

const useGetEdit = (requestOptions, queryOptions = { queryKey: 'getEdit' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getEdit(_id, requestOptions),
	});
};

const usePostCourse = (requestOptions, mutateOptions = { queryKey: 'postCourse' }) => {
	return useMutation(data => postCourse(data, requestOptions), {
		mutateOptions,
	});
};

const usePutCourse = (requestOptions, mutateOptions = { queryKey: 'putCourse' }) => {
	return useMutation((_id, data) => putCourse(_id, data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCourse,
	getEdit,
	postCourse,
	putCourse,
	useGetCourse,
	useGetEdit,
	usePostCourse,
	usePutCourse,
};
