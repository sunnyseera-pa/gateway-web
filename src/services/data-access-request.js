import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { deleteRequest, getRequest, patchRequest, postRequest, putRequest } from '../utils/requests';

const getDataAccessRequests = options => {
	return getRequest(`${apiURL}/data-access-request`, options);
};

const getDataAccessRequest = (_id, options) => {
	return getRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const postDataAccessRequest = (_id, data, options) => {
	return postRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const putDataAccessRequest = (_id, data, options) => {
	return putRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const patchDataAccessRequest = (_id, data, options) => {
	return patchRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const deleteDataAccessRequest = (_id, options) => {
	return deleteRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const useGetDataAccessRequests = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getDataAccessRequests(requestOptions),
	});
};

const useGetDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => getDataAccessRequest(id, requestOptions),
	});
};

const usePostDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'postDataAccessRequest' }) => {
	return useMutation((id, data) => postDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePutDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'putDataAccessRequest' }) => {
	return useMutation((id, data) => putDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePatchDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'patchDataAccessRequest' }) => {
	return useMutation((id, data) => patchDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const useDeleteDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'deleteDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => deleteDataAccessRequest(id, requestOptions),
	});
};

export default {
	getDataAccessRequests,
	getDataAccessRequest,
	postDataAccessRequest,
	putDataAccessRequest,
	patchDataAccessRequest,
	deleteDataAccessRequest,
	useGetDataAccessRequests,
	useGetDataAccessRequests,
	useGetDataAccessRequest,
	usePostDataAccessRequest,
	usePutDataAccessRequest,
	usePatchDataAccessRequest,
	useDeleteDataAccessRequest,
};
