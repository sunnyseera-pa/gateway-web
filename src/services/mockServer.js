import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';
import mswSearch from './search/mockMsw';
import mswGetLocations from './locations/mockMsw';
import mswGetContributors from './contributors/mockMsw';

import translations from '../../public/locales/en-GB/translation.json';

const mswGetEnTranslations = rest.get(`http://localhost/locales/en/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetEnGbTranslations = rest.get(`http://localhost/locales/en-GB/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetIcon = rest.get(`/images/Application_approved.svg`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.text('<svg />'));
});

const handlers = [
	...mswDatasets,
	...mswDatasetOnboarding,
	...mswPostDatasetActivityLog,
	// ...mswSearch,
	...mswGetLocations,
	...mswGetContributors,
	mswGetEnTranslations,
	mswGetEnGbTranslations,
	mswGetIcon,
];

export const server = setupServer(...handlers);
