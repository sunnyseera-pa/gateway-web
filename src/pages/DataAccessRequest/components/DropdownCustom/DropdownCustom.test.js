import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import DropdownCustom from './DropdownCustom';
import { server } from '../../../../services/mockServer';
import '@testing-library/jest-dom/extend-expect';

const dropdownCustomProps = {
	name: 'safepeopleprimaryapplicantfullname',
	id: 'safepeopleprimaryapplicantfullname',
	questionSetId: 'primaryapplicant',
	labelId: 'safepeopleprimaryapplicantfullname-label',
	value: 'Ciara Ward',
	disabled: false,
	required: true,
	readOnly: false,
	classes: {
		form: 'dar-form',
		select: 'form-control',
		typeaheadCustom: 'form-control',
		datePickerCustom: 'form-control',
		question: 'form-group',
		questionWrap: 'question-wrap',
		input: 'form-control',
		button: 'btn btn-primary',
		radioListItem: 'dar__radio--item',
		radioList: 'dar__radio--list list-group',
		checkboxInput: 'checkbox list-group',
		checkboxListItem: 'dar__check--item ',
		checkboxList: 'dar__check list-group',
		controlButton: 'btn btn-primary pull-right',
		backButton: 'btn btn-default pull-left',
		errorMessage: 'alert alert-danger',
		alertWrap: 'alert-wrap',
		buttonBar: 'button-bar hidden',
		actionControl: 'action-control',
		nested: 'nested',
		toolTip: 'toolTip',
		toolTipTop: 'toolTip-top',
		toolTipText: 'toolTipText',
		actionCount: 'actionCount',
		actionNotifications: 'action-notifications',
	},
	applicationId: 123,
};
let wrapper;
let input;
describe('Given the DropdownCustom component', () => {
	beforeAll(() => {
		server.listen();
	});

	afterEach(() => {
		server.resetHandlers();
	});

	afterAll(() => {
		server.close();
	});

	describe('When it is rendered', () => {
		beforeAll(() => {
			server.listen();
			wrapper = render(<DropdownCustom {...dropdownCustomProps} />, {
				wrapper: Providers,
			});
			input = document.querySelector('.rbt-input-main');
			fireEvent.click(input);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then name should be rendered for 3 contributors', async () => {
			await waitFor(() => {
				expect(wrapper.getByTestId('darContributorDropdownName-0')).toHaveTextContent('Ciara Ward');
				expect(wrapper.getByTestId('darContributorDropdownName-1')).toHaveTextContent('Paul McCafferty');
				expect(wrapper.getByTestId('darContributorDropdownName-2')).toHaveTextContent('Ciara Test');
			});
		});

		it('Then email should be rendered for only the logged in users email displayed', async () => {
			await waitFor(() => {
				expect(wrapper.getByTestId('darContributorDropdownEmail-0')).toHaveTextContent('ciara.ward@paconsulting.com');
				expect(wrapper.getByTestId('darContributorDropdownEmail-1')).toHaveTextContent('Email address cannot be shared');
				expect(wrapper.getByTestId('darContributorDropdownEmail-2')).toHaveTextContent('Email address cannot be shared');
			});
		});

		it('Then renders with the correct organisations shown and hidden', async () => {
			await waitFor(() => {
				expect(wrapper.getByTestId('darContributorDropdownOrganisation-0')).toHaveTextContent('test');
				expect(wrapper.getByTestId('darContributorDropdownOrganisation-1')).toHaveTextContent('PA Consulting');
				expect(wrapper.getByTestId('darContributorDropdownOrganisation-2')).toHaveTextContent('Organisation is hidden');
			});
		});
	});
});
