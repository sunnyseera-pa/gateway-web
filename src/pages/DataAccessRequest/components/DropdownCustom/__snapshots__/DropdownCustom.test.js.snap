// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DropdownCustom component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="ui-LayoutBox css-1sivg29"
  >
    <div
      class="css-1r9m6uy-Input form-group"
    >
      <label
        class="css-prhs3j-Input form-label"
      />
      <div
        class="ui-Input ui-Typeahead css-kuqv2w-Input input-group"
      >
        <div
          class="rbt"
          style="outline: none; position: relative;"
          tabindex="-1"
        >
          <div
            style="display: flex; flex: 1; height: 100%; position: relative;"
          >
            <input
              aria-autocomplete="both"
              aria-expanded="true"
              aria-haspopup="listbox"
              aria-owns="prepopulate"
              autocomplete="off"
              class="rbt-input-main form-control rbt-input  focus"
              role="combobox"
              type="text"
              value="Ciara Ward"
            />
            <input
              aria-hidden="true"
              class="rbt-input-hint"
              readonly=""
              style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.35); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: solid solid solid solid; border-width: 2px 2px 2px 2px; font-size: 13px; height: 40px; line-height: normal; padding: 1px 1px 1px 1px;"
              tabindex="-1"
              value=""
            />
          </div>
          <ul
            aria-label="menu-options"
            class="rbt-menu dropdown-menu show"
            id="prepopulate"
            role="listbox"
            style="position: absolute; display: block; max-height: 300px; overflow: auto; top: 0px; left: 0px; transform: translate3d(NaNpx, NaNpx, 0); will-change: transform; width: 0px;"
          >
            <li
              class="disabled"
              role="option"
            >
              <a
                class="dropdown-item disabled"
                href="#"
              >
                No matches found.
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
